### ISSUE TITLE
Deployer composer installation failed

### ISSUE THREAD

**Issue by nicolastorgato** [deployphp/deployer#3104]
Hi, 
I'm using Windows 10, Laravel 9.2 and PHP 8.0.2. 
I'm trying to install Deployer on a fresh Laravel project, using: `composer require --dev deployer/deployer`

I'm always getting this error:
```
 Problem 1
    - deployer/deployer[v4.3.0, ..., 4.x-dev] require monolog/monolog ^1.21 -> found monolog/monolog[1.21.0, ..., 1.x-dev] but the package is fixed to 2.5.0 (lock file version) by a partial update and that 
version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires deployer/deployer ^4.3 -> satisfiable by deployer/deployer[v4.3.0, ..., 4.x-dev].
```

And what's the difference between installing Deployer with composer in the project folder or instead downloading "deployer.phar"?

**Comment by valguss** [comment]
Looks like you're trying to install deployer v4
Latest release is 6 and we're into RC for 7

The main difference is that if you install via composer, then every one on the project will be using the same version

**Comment by nicolastorgato** [comment]
Hi valguss, 
I mean, I just followed the documentation where the only command you guys provided is `composer require --dev deployer/deployer`...
What am I supposed to do? I tried to install the RC version using `composer require --dev deployer/deployer:v7.0.0-rc.8` and it works.

Now I'm trying to enter the following command `vendor/bin/deployer.phar init`, but VSCode prompts me to open the .phar file or, if I try on GitBash, I receive this error:

```
PHP Warning:  Phar::mapPhar(C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code): Failed to open stream: No such file or directory in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code on line 2

Warning: Phar::mapPhar(C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code): Failed to open stream: No such file or directory in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code on line 2
PHP Fatal error:  Uncaught PharException: unable to open phar for reading "C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code" in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code:2
Stack trace:
#0 C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code(2): Phar::mapPhar('deployer.phar')
#1 C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21): eval()
#2 {main}
  thrown in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code on line 2

Fatal error: Uncaught PharException: unable to open phar for reading "C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code" in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code:2
Stack trace:
#0 C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code(2): Phar::mapPhar('deployer.phar')
#1 C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21): eval()
#2 {main}
  thrown in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code on line 2
```

I'm using Windows 10, maybe it has to do with this...

**Comment by valguss** [comment]
So if it's installed OK via composer, then the phar file should be in the `vendor/bin` directory. Double check it actually is

I haven't got any experience with running this through WIndows but this also feels like an environment issue. It might be worth removing deployer from composer and installing deployer like this:

https://deployer.org/docs/6.x/installation

**Comment by nicolastorgato** [comment]
It's in that directory but it doesn't work. I even tried with Docker but it doesn't work either...
At this point, I think I will find another way or another tool. 
Thanks anyway!

**Comment by cracki** [comment]
> Hi, I'm using Windows 10, Laravel 9.2 and PHP 8.0.2. I'm trying to install Deployer on a fresh Laravel project, using: `composer require --dev deployer/deployer`
> 
> I'm always getting this error:
> 
> ```
>  Problem 1
>     - deployer/deployer[v4.3.0, ..., 4.x-dev] require monolog/monolog ^1.21 -> found monolog/monolog[1.21.0, ..., 1.x-dev] but the package is fixed to 2.5.0 (lock file version) by a partial update and that 
> version does not match. Make sure you list it as an argument for the update command.
>     - Root composer.json requires deployer/deployer ^4.3 -> satisfiable by deployer/deployer[v4.3.0, ..., 4.x-dev].
> ```
> 
> And what's the difference between installing Deployer with composer in the project folder or instead downloading "deployer.phar"?

same problem :(

> Hi valguss, I mean, I just followed the documentation where the only command you guys provided is `composer require --dev deployer/deployer`... What am I supposed to do? I tried to install the RC version using `composer require --dev deployer/deployer:v7.0.0-rc.8` and it works.
> 
> Now I'm trying to enter the following command `vendor/bin/deployer.phar init`, but VSCode prompts me to open the .phar file or, if I try on GitBash, I receive this error:
> 
> ```
> PHP Warning:  Phar::mapPhar(C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code): Failed to open stream: No such file or directory in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code on line 2
> 
> Warning: Phar::mapPhar(C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code): Failed to open stream: No such file or directory in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code on line 2
> PHP Fatal error:  Uncaught PharException: unable to open phar for reading "C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code" in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code:2
> Stack trace:
> #0 C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code(2): Phar::mapPhar('deployer.phar')
> #1 C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21): eval()
> #2 {main}
>   thrown in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code on line 2
> 
> Fatal error: Uncaught PharException: unable to open phar for reading "C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code" in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code:2
> Stack trace:
> #0 C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code(2): Phar::mapPhar('deployer.phar')
> #1 C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21): eval()
> #2 {main}
>   thrown in C:\xampp\htdocs\LARAVEL\deployer-error-test\vendor\bin\deployer.phar(21) : eval()'d code on line 2
> ```
> 
> I'm using Windows 10, maybe it has to do with this...

try this:
` php vendor/bin/deployer.phar init`
